### DISCUSSION TITLE
how to make POST api request by click on Save Button

### DISCUSSION THREAD

**Discussion by kyvox-dev** [refinedev/refine#4325] [Q&A] [answered]
I have to call this  POST api  "users/edit/:id" with refine inferencer

above API config contain dynamic id in  param and other data in body

![Screenshot (9)](https://github.com/refinedev/refine/assets/126939698/364da468-3010-48a2-acf0-cfaa21917e55)

**Comment by salihozdemir** [comment]
Hey @kyvox-dev ,

You can use meta to send custom variables to your data provider.

You can check this to learn how to do it: https://refine.dev/docs/tutorial/understanding-dataprovider/create-dataprovider/#meta-usage
Btw, the inferencer has been developed for a development environment. So, you can't do this on Inferencer. However, you can copy the generated code and you can customize it.

**Comment by alicanerdurmaz** [comment, marked as answer]
Hey @kyvox-dev, We created another solution and example for you. 
you can use `useCustomMutation` with form data and you can send your request with any method.
https://codesandbox.io/s/damp-snowflake-qog2fn?file=/src/pages/posts/edit.tsx:938-1009

**Reply by alicanerdurmaz** [reply]
@kyvox-dev I updated exmaple to use `useUpdate` instead of `useCustomMutation` 
because `useUpdate` automatically invalidates resource
https://codesandbox.io/s/hardcore-browser-trdro5?file=/src/pages/posts/edit.tsx:843-914
